lib/chef/resource/windows_user_privilege.rb

```diff
@@ -0,0 +1,159 @@
+
+      introduced "16.0"
+
+      property :principal, String,
+        description: "An optional property to add the user to the given privilege. Use only with add and remove action.",
+        name_property: true
+
+      property :users, Array,
+        description: "An optional property to set the privilege for given users. Use only with set action."
+
+      property :privilege, [Array, String],
+        description: "Privilege to set for users.",
+        required: true,
+        callbacks: {
+           "Option privilege must include any of the: #{privilege_opts}" => lambda {
+             |v| v.is_a?(Array) ? (privilege_opts & v).size == v.size : privilege_opts.include?(v)
+           },
+         }
+
+      property :sensitive, [TrueClass, FalseClass], default: true
+
+      load_current_value do |new_resource|
+        unless new_resource.principal.nil?
+          privilege Chef::ReservedNames::Win32::Security.get_account_right(new_resource.principal) unless new_resource.action.include?(:set)
+        end
+      end
+
+      action :add do
+        ([*new_resource.privilege] - [*current_resource.privilege]).each do |user_right|
+          converge_by("adding user privilege #{user_right}") do
+            Chef::ReservedNames::Win32::Security.add_account_right(new_resource.principal, user_right)
+          end
+        end
+      end
+
+      action :set do
+        if new_resource.users.nil? || new_resource.users.empty?
+          raise Chef::Exceptions::ValidationFailed, "Users are required property with set action."
+        end
+
+        users = []
+
+        # Getting users with its domain for comparison
+        new_resource.users.each do |user|
+          user = Chef::ReservedNames::Win32::Security.lookup_account_name(user)
+          users << user[1].account if user
+        end
+
+        new_resource.privilege.each do |privilege|
+          accounts = Chef::ReservedNames::Win32::Security.get_account_with_user_rights(privilege)
+
+          # comparing the existing accounts for privilege with users
+          unless users == accounts
+            accounts.each do |account|
+              converge_by("removing user #{account[1]} from privilege #{privilege}") do
+                Chef::ReservedNames::Win32::Security.remove_account_right(account[1], privilege)
+              end
+            end
+
+            new_resource.users.each do |user|
+              converge_by("adding user #{user} to privilege #{privilege}") do
+                Chef::ReservedNames::Win32::Security.add_account_right(user, privilege)
+              end
+            end
+          end
+        end
+      end
+
+      action :remove do
+        curr_res_privilege = current_resource.privilege
+        new_res_privilege = new_resource.privilege
+
+        new_res_privilege = [] << new_res_privilege if new_resource.privilege.is_a?(String)
```

Comment:
that should just be `new_res_privilege = Array(new_resource.new_res_privilege)`

Reply:
Will update this.

Reply:
@lamont-granquist Removed this line and handled it through coerce as it required to covert in array for other actions too. Thank you .